app/src/main/java/com/android/voyageur/model/user/UserViewModel.kt

```diff
@@ -193,45 +206,111 @@ open class UserViewModel(
         onSuccess = { getSentFriendRequests() },
         onFailure = {})
   }
+
   /**
    * Adds a contact to the current user's contact list.
    *
    * @param userId The ID of the user to add as a contact.
+   * @param friendRequestId The ID of the friend request to delete after adding the contact.
    */
   fun addContact(userId: String, friendRequestId: String) {
-    val currentUser = user.value ?: return
-    val updatedContacts = currentUser.contacts + userId // Creates a new list
-    val updatedUser = currentUser.copy(contacts = updatedContacts)
-    updateUser(updatedUser)
-    deleteFriendRequest(friendRequestId)
+    val contacts = user.value?.contacts?.toMutableSet()
+    val newUser = user.value!!.copy()
+    contacts?.add(userId)
+    newUser.contacts = contacts?.toList().orEmpty()
+    if (user.value != null) {
+      updateUser(newUser)
+      // Deletes Friend Request since the user has been added as a contact
+      deleteFriendRequest(friendRequestId)
+    }
```

Comment:
`if (user.value != null)` is redundant after `val newUser = user.value!!.copy()`. Can we instead do something like     `val currentUser = user.value ?: return` at the beginning of the function?

Reply:
Thank you for the suggestion! You're absolutely right—if (user.value != null) is redundant after val newUser = user.value!!.copy(), as the !! operator already guarantees that user.value is not null. Using val currentUser = user.value ?: return at the beginning of the function simplifies the code and avoids repetitive null checks.